tests/sorttest.lobster

```diff
@@ -0,0 +1,33 @@
+import std
+
+// FIXME why is the `0` literal typed
```

Comment:
What is the problem? `0` is an `int` in most statically typed languages.

Reply:
Well the problem is typing: you can't use `sum(xs)` on an `xs` of type `[float]` because the `0` initial accumulator to `fold` is an int.

Hence my need to define `sumf` here which starts a `fold` out with `0.0` instead.

Reasons like this are why, e.g. in Go, literals and even numeric constants (as in `const foo = 42`) untyped; if instead you want a typed constant, you do `const foo int = 42` or `const foo float64 = 42`.

Reply:
While I agree that it is inconvenient here, typically it is the right thing to do, and what most static languages do. Go is again the odd one out here (as it is with a lot of things). Being able to get the int or float version of a function by just using a constant is usually very convenient.

None of these languages have the "automatic genericity" of Lobster though, which makes this a special case. Really the best way to solve this would be to allow explicit type parameters, e.g. `def sumf<T>(xs:[T]): return fold(xs, T(0)): _x + _y` or similar.

For now, my solution would be to add a starting value to `sum` rather than define a second version.

Note `weighted_pick` right above there in `std.lobster` has the exact same problem. I guess its time for explicit template parameters :)

Reply:
https://github.com/aardappel/lobster/commit/a9011897b9870c42e468ae75b44b6b50c9f7c3ab